CMakeLists.txt

```diff
@@ -172,7 +172,16 @@ else()
 	set(LUAJIT_INCLUDE "${LUAJIT_SRC}")
 	if(NOT WIN32)
 		set(LUAJIT_LIB "${LUAJIT_SRC}/libluajit.a")
- 		if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "ppc64le")
+		if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "aarch64")
+		ExternalProject_Add(luajit
+			GIT_REPOSITORY "https://github.com/cbaylis/luajit-aarch64.git"
```

Comment:
The current master of Luajit supports aarch64.
I've been working on the same thing for Falco here
https://github.com/falcosecurity/falco/pull/1442/files#diff-d36917b66235e5b6e2421e36f5fd94f0b6b09454f0279b0c6549aa9bf6c93761R20-R21 .

Also I believe that it can replace moonjit for ppc64le now

Reply:
> https://github.com/falcosecurity/falco/pull/1442/files#diff-d36917b66235e5b6e2421e36f5fd94f0b6b09454f0279b0c6549aa9bf6c93761R20-R21 .

The version of Luajit (pointed out here) doesn't look to be compatible with Sysdig. I ran the builds on AArch64/x86 and both the builds got failed. Please have a look at below logs-
x86_64 Logs: https://travis-ci.com/github/mpmkp2020/sysdig/jobs/427790745
AArch64 Logs: https://travis-ci.com/github/mpmkp2020/sysdig/jobs/427790744

Also, the same version of Luajit doesn't support ppc64le. Please have a loot at below logs
Ppc64le Logs: https://travis-ci.com/github/mpmkp2020/sysdig/jobs/427901220

Reply:
For aarch/x86 the error is:

```
I/home/travis/build/mpmkp2020/sysdig/build/userspace/libsinsp -I/home/travis/build/mpmkp2020/sysdig/build/tbb-prefix/src/tbb/include  -Wall -ggdb   -std=c++0x -D_DEBUG   -o CMakeFiles/sinsp.dir/chisel.cpp.o -c /home/travis/build/mpmkp2020/sysdig/userspace/libsinsp/chisel.cpp
/home/travis/build/mpmkp2020/sysdig/userspace/libsinsp/chisel.cpp:135:1: error: elements of array â€˜const luaL_reg ll_sysdig []â€™ have incomplete type
 };
 ^
/home/travis/build/mpmkp2020/sysdig/userspace/libsinsp/chisel.cpp:135:1: error: storage size of â€˜ll_sysdigâ€™ isnâ€™t known
/home/travis/build/mpmkp2020/sysdig/userspace/libsinsp/chisel.cpp:147:1: error: elements of array â€˜const luaL_reg ll_chisel []â€™ have incomplete type
 };
 ^
/home/travis/build/mpmkp2020/sysdig/userspace/libsinsp/chisel.cpp:147:1: error: storage size of â€˜ll_chiselâ€™ isnâ€™t known
/home/travis/build/mpmkp2020/sysdig/userspace/libsinsp/chisel.cpp:157:1: error: elements of array â€˜const luaL_reg ll_evt []â€™ have incomplete type
 };
```

That's the reason why when we import Sysdig into Falco we patch it with this patch: https://github.com/falcosecurity/falco/blob/build/aarch64/cmake/modules/sysdig-repo/patch/luajit.patch

That is the same thing that it's done here https://github.com/draios/sysdig/blob/travis-ci-aarch64/CMakeLists.txt#L188 but it will be not necessary if in this PR you apply the patch and commit it (we don't need to patch on the fly anymore if all the Luajit implementations need the patched types)

So I would say that we need to:

- Apply the patch https://github.com/falcosecurity/falco/blob/build/aarch64/cmake/modules/sysdig-repo/patch/luajit.patch on Sysdig and commit it
- Use the luajit version I pointed out earlier for x86 and aarch64
- Continue using moonjit for ppc64le

Wdyt?

Reply:
Yes, I agree with you. Made the changes as per your suggestions. Could you please review the changes.

Reply:
Thanks! One last thing, since now the code has the references to the new LuaJIT structs we need to remove the `PATCH_COMMAND` at lines 179 and 189 of `CMakeLists.txt`. 

We also need to use the new version of LuaJIT for the WIN32 else here https://github.com/draios/sysdig/pull/1706/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR206-R214

Thanks a lot, this is very important work for the future of this project on arm! :)

Reply:
@fntlnz, I have removed the `PATCH_COMMAND` from the lines you pointed out and pushed the changes. Sorry, due to lack or resources, I can not work on WIN32.

Reply:
@fntlnz can you please review the changes?

Reply:
I have to get my hands on a windows builder and see if we break that. I'll come back soon

Reply:
Thanks for your persistence on this!